web_src/js/modules/fetch.js

```diff
@@ -0,0 +1,38 @@
+import {isObject} from '../utils.js';
+
+const {csrfToken} = window.config;
+
+// fetch wrapper, use below method name functions and the `data` option to pass in data
+// which will automatically set an appropriate content-type header. For json content,
+// only object and array types are currently supported.
+function request(url, {headers, data, body, ...other} = {}) {
+  let contentType;
+  if (!body) {
+    if (data instanceof FormData) {
+      contentType = 'multipart/form-data';
+      body = data;
+    } else if (data instanceof URLSearchParams) {
+      contentType = 'application/x-www-form-urlencoded';
```

Comment:
`multipart/form-data` should be always the correct content type.

While I do not see why it should support `URLSearchParams`, it should mainly used for URL query string, but not a POST request.